File: src/fibo.py
import time, sys

#cache = {}	# python dictionary
cache = dict()

def fiboCached(n):
	if (n in cache):	# if (cache[n] is not None) will throw an exception in python
		return cache[n]
	if (n == 1):
		return 1
	if (n == 0):
		return 0
	
	cache[n] = fibo(n-1) + fibo(n-2)
	return cache[n]

def fibo(n):

	if (n == 1):
		return 1
	if (n == 0):
		return 0
	
	return fibo(n-1) + fibo(n-2)

def main():
	now = time.time()

	if (len(sys.argv) > 1):
		arg = int(sys.argv[1])
	else:
		arg = 30

	print(fibo(arg));
	print 'Python Fibonacci ' + str(arg) + ': Time elapsed: {0} milli seconds'.format((time.time() - now)*1000)

if __name__ == "__main__":
	main()

File: src/fibo.c
#include <time.h>
#include <stdio.h> 
#include <sys/time.h>

int fibo(n) {
	if (n == 0) {
		return 0;
	}

	if (n == 1) {
		return 1;
	}

	return fibo(n-1) + fibo(n-2);
}

int main(int argc, char *argv[]) {
	struct timeval start, finish;
	clock_t begin, end, elapsed;
	
	gettimeofday( &start, NULL );

	int arg = 30;
	if (argc > 1) {
		arg = atoi(argv[1]);	// ASCII to integer
	}
	// begin = clock();	// clock measures CPU time rather than 'real' time
	printf("%d\n", fibo(arg));

    gettimeofday( &finish, NULL );
	// end = clock();
	
	elapsed = (double)(end-begin) / CLOCKS_PER_SEC;
	int timeuse = (1000000 * ( finish.tv_sec - start.tv_sec ) + finish.tv_usec -start.tv_usec) / 1000;

    printf("C Fibonacci %d---> Time used: %d ms\n", arg, timeuse);
//	printf("Time elapsed: %ld s\n", elapsed);
}